src/challenges.js

```diff
@@ -28,18 +36,22 @@ const cacheFunction = cb => {
 
 /* ======================== Recursion Practice ============================ */
 const reverseStr = str => {
-  // reverse str takes in a string and returns that string in reversed order
-  // The only difference between the way you've solved this before and now is that you need to do it recursivley!
+  var first = str[0];
+  var last = str[str.length -1];
+  var str1 = reverseStr(str.substring(1, str.length - 1));
+  return last + str1 + first;
 };
+// based on https://stackoverflow.com/questions/4859208/recursive-string-reversal-function-in-javascript
 
 const checkMatchingLeaves = obj => {
   // return true if every property on `obj` is the same
   // otherwise return false
 };
 
 const flatten = elements => {
-  // Flattens a nested array (the nesting can be to any depth).
-  // Example: flatten([1, [2], [3, [[4]]]]); => [1, 2, 3, 4];
+  each(elements, element => {       // call each function to iterate over the array
+    elements = flatten([].concat([], ...elements));   // flatten by breaking out nested arrays as we recursively call flatten function
```

Comment:
Wow, That is an amazing solution! I'm going to have to look into this and see if there are any edge cases that could potential pop up, for now, I'd tally this one up as my favorite solution to flatten.